### ISSUE TITLE
Disappearing cursor over match paren

### ISSUE THREAD

**Issue by cideM** [tyrannicaltoucan/vim-deep-space#10]
This is Alacritty, Tmux, Neovim on MacOs but I have the same issue outside of Tmux and also on Arch Linux with both Alacritty and Termite. I have this in a bunch of color schemes and there are some bug reports about this on the base16 repo as well. If you would like more info let me know. Note that I am using dynamic cursor colors (meaning cursor color is adjusted based on background).

Also, the cursor is sitting on the first paren (I have RainbowColors active in the screenshot but it doesn't change the issue)

<img width="262" alt="screen shot 2018-04-27 at 17 25 41" src="https://user-images.githubusercontent.com/4246921/39370776-10a55c1a-4a40-11e8-90a6-d509110c7a5b.png">

**Comment by tyrannicaltoucan** [comment]
Hmm... I can't reproduce this at all on iTerm:

<img width="472" alt="matchparens" src="https://user-images.githubusercontent.com/18494520/39450637-a347f80e-4c91-11e8-9006-9dedd36ab208.png">

I think I just need to adjust the `MatchParen` foreground. Try overriding `MatchParen` by adding the following after setting the color scheme in your `init.vim`:

```vim
hi MatchParen guibg=#c47ebd guifg=#51617d
```

Does this make the cursor visible again?

**Comment by cideM** [comment]
Yeah that works. Anything that's not just `reverse` or manually reversing the colors work. I just tried it without dynamic cursor colors in both alacritty and termite and that also solves the problem. Now I am not sure if maybe the dynamic cursor adjustment of the terminal is confused by the MatchParen color change. In other words, I'm not sure if the terminals behave in the expected way here (although I'd err on the side of yes they do). What all color schemes that give me these issues have in common is that `MatchParen` uses the background from `Normal` (or almost the background, here it's `gray1` and `gray2`) or just `reverse`). One color scheme where it works flawlessy is [seoul256](https://github.com/junegunn/seoul256.vim/blob/master/colors/seoul256.vim#L292)

Bottom line: if there is `MatchParen` color combo that you're okay with I'd be super happy about that :) but I'd understand if you consider this an edge case that people with terminals with such cursor adjustments have to live with

**Comment by tyrannicaltoucan** [comment]
I'll just change it. There are way too many terminals with this behavior to call it an edge case.
I should have some time this weekend to mull it over and come up with something.

**Comment by elquimista** [comment]
I have a similar issue, but this is more like an upside-down cursor problem. See screenshot:
<img width="270" alt="screen shot 2018-11-20 at 12 50 28 pm" src="https://user-images.githubusercontent.com/14348482/48802046-e1266300-ecc2-11e8-84e9-35be6ad0560c.png">
My cursor is actually on the opening curly brace in the screenshot (col 14), but it's misleading that it looks as if cursor is on the closing curly brace (col 34). With other color schemes, I don't have this issue.
